src/app/modules/shared-components/components/progress-level/progress-level.component.html

```diff
@@ -0,0 +1,6 @@
+<div class="progress-level-container">
+  <ng-container *ngFor="let i of totalArray()">
```

Comment:
You cannot call a function from your html, otherwise angular call it each time there is a tiny unrelated change on the page. Add a console log in your function and you'll see.

Reply:
ok now I store the array in the component, and initialize it in the constructor.